http/http/src/main/java/io/helidon/http/PathMatchers.java

```diff
@@ -60,9 +61,22 @@ public static PathMatcher exact(String pathToMatch) {
      *
      * @param pathToMatch the path must be prefixed by the provided path
      * @return prefix match path matcher
+     * @deprecated use {@link #prefix(String, String)} instead to preserve original pattern
      */
+    @Deprecated(forRemoval = true, since = "4.3.0")
     public static PathMatcher prefix(String pathToMatch) {
-        return new PrefixPathMatcher(fixPrefix(pathToMatch));
+         return prefix(pathToMatch, null);
+    }
+
+    /**
+     * Prefix match path matcher that stores original path pattern.
+     *
+     * @param pathToMatch matching path such as "/foo"
+     * @param pathPattern complete path pattern such as "/foo/*"
+     * @return prefix match path matcher
+     */
+    public static PathMatcher prefix(String pathToMatch, String pathPattern) {
+        return new PrefixPathMatcher(fixPrefix(pathToMatch), pathPattern);
```

Comment:
There should be a check for not-null for both parameters.
I am not sure I understand why the second parameter is added.

Reply:
Yes, this seems to work fine, and `/foo/*/*` is handled as a `PatternPathMatcher`. See updates and new test.